player/screenshot.c

```diff
@@ -127,6 +127,30 @@ static void append_filename(char **s, const char *f)
     talloc_free(append);
 }
 
+static void trim_invalid_utf8(char *s, size_t len)
+{
+    for (int i = 1; i <= 3; i++) {
+        if (bstr_parse_utf8_code_length(s[len-i]) > i) {
```

Comment:
will underflow if len < 3

Reply:
Side-stepped this when inlining by making `truncate_long_base_filename` take `strlen(extension)` rather than `max_utf8_bytes`.